### PR TITLE
Fixed counting of faces for 1d.

### PR DESCRIPTION
**Pull request by luca-heltai**
We now offer face iterators for 1D triangulations, but the face count is inconsistent.

This fixes it.

### CODE REVIEW

**Comment by masterleinad** [comment]
The indentation is off. :wink:

**Comment by luca-heltai** [comment]
Actually the indentation is ok. Travis is testing the wrong commit. I have no idea how to force it to restart. I had force pushed, and this confused Travis.

**Comment by luca-heltai** [comment]
I tried rebasing... This should work...

File: include/deal.II/grid/tria.h [hunk 1]
```diff
@@ -3073,16 +3073,16 @@ class Triangulation : public Subscriptor
 
   /**
    * Return the total number of used faces, active or not.  In 2D, the result
-   * equals n_lines(), while in 3D it equals n_quads(). Since there are no
-   * face objects in 1d, the function returns zero in 1d.
+   * equals n_lines(), in 3D it equals n_quads(), while in 1D it equals
+   * the number of vertices.
```
Comment (bangerth) [inline include/deal.II/grid/tria.h]: say "used vertices" to be consistent with the `get_vertices()` and `get_used_vertices()` functions

Comment (luca-heltai) [reply include/deal.II/grid/tria.h]: mmm...  this is not the number of used vertices, though. It is the total number of vertices (i.e., all faces, including inactive faces, which, in 1D, are inactive vertices).

Comment (bangerth) [reply include/deal.II/grid/tria.h]: No, that's not quite right. All used vertices are active, and so `n_face()` should return the same as `n_active_faces()`. But what the `get_vertices()` function returns is an array of point locations of which only some are actually used. The difference between used and unused is the difference between `cell_iterator` and `raw_cell_iterator`.

The point I'm trying to make is that of course the returned number of faces in 1d is equal to the number of vertices *that exists as part of the (logical) mesh*, but it is a subset of the vertices *stored in the data structure `class Triangulation`*.

File: source/grid/tria.cc [hunk 2]
```diff
@@ -12520,7 +12520,7 @@ Triangulation<dim, spacedim>::n_faces() const
   switch (dim)
     {
       case 1:
-        return 0;
+        return n_vertices();
       case 2:
         return n_lines();
```
Comment (bangerth) [inline source/grid/tria.cc]: This is the implementation of the `n_faces()` function. Note how in 2d, you get `n_lines()`, which is documented as follows:
```
   * Return the total number of used lines, active or not.
```
On the other hand, `n_vertices()` is documented like this:
```
   * Return the total number of vertices.  Some of them may not be used, which
   * usually happens upon coarsening of a triangulation when some vertices are
   * discarded, but we do not want to renumber the remaining ones, leading to
   * holes in the numbers of used vertices.  You can get the number of used
   * vertices using @p n_used_vertices function.
```
So `n_vertices()` does not take the *used* status into account.

What you want to call here is `n_used_vertices()`, which is described as follows:
```
   * Return the number of vertices that are presently in use, i.e. belong to
   * at least one used element.
```

File: source/grid/tria.cc [hunk 3]
```diff
@@ -12538,6 +12538,8 @@ Triangulation<dim, spacedim>::n_raw_faces() const
 {
   switch (dim)
     {
+      case 1:
+        return n_vertices();
       case 2:
         return n_raw_lines();
```
Comment (bangerth) [inline source/grid/tria.cc]: This is `n_raw_faces()`, where you are calling the correct function.

File: source/grid/tria.cc (hunk 2 above)
Comment (luca-heltai) [reply source/grid/tria.cc]: Ahhh.  Now I see your point. Sorry!

👍

**Pull request merged**